lib/streamer/cbor_streamer.go

```diff
@@ -0,0 +1,199 @@
+package streamer
+
+import (
+	"io"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/ugorji/go/codec"
+)
+
+var _ Mux = &CborMux{}
+
+type CborMux struct {
+	// the fd tip handles append; reads all use ReadAt and track their offsets individually
+	file  *os.File
+	codec *codec.Encoder
+	wmu   sync.Mutex
+}
+
+type cborMuxRow struct {
+	Label int    `json:"l"`
+	Msg   []byte `json:"m,omitempty"`
+	Sig   int    `json:"x,omitempty"` // 1->closed
+}
+
+func CborFileMux(filePath string) Mux {
+	file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_EXCL, 0755)
+	if err != nil {
+		panic(errors.IOError.Wrap(err))
+	}
+	file.Write([]byte{codec.CborStreamArray})
```

Comment:
There's no requirement that processes close their starting file descriptors all at once.  So inevitably, they don't, and Something is going to rely on that.  I remember finding this out the hard way while working on siphon at some point, though I don't think I was decent enough at the time to write tests or examples.

I guess we don't really *need* the array close flag in order to be valid cbor, anyway?  Yeah, check out around the `dec.MustDecode(&reheated)` line in the tests... apparently that's parsing validly.
Something like that would work, but adding new labels to the open set while other goroutines are already allowed to be closing them is going to make a big picture race, so either the mux interface should start looking like a factory to make that clear, or we're gonna get a rather sharp "safe" usage path where the caller must make sure to open all appenders before possibly closing any of them.